### ISSUE TITLE
Attempt to document installation of PHP 7.4 and necessary extensions on Ubuntu 22.04

### ISSUE THREAD

**Issue by pako81** [owncloud/docs-server#252]
## WHAT Needs to be Documented?

Attempt to document installation of PHP 7.4 and necessary extensions on Ubuntu 22.04. 

1. Install SURY APT Repository: to install PHP 7.4 on Ubuntu 22.04 we will need to install SURY, third party repositories that provides PHP packages (we are going to use the Focal one as Jammy one is not available yet):

```
sudo add-apt-repository ppa:ondrej/php -y
sed -i 's/jammy/focal/' /etc/apt/sources.list.d/ondrej-ubuntu-php-jammy.list
sudo apt update
```

2. Running below command to install PHP7.4:

`sudo apt install php7.4`

returns an error:

```
The following packages have unmet dependencies:
 libapache2-mod-php7.4 : Depends: php7.4-cli but it is not going to be installed
                                             Depends: libssl1.1 (>= 1.1.0) but it is not installable
                                             Recommends: apache2 but it is not going to be installed
 php7.4-common : Depends: libssl1.1 (>= 1.1.0) but it is not installable
```

Problem —> In Ubuntu 22.04 `libssl1.1` has been superseded by `libssl3` and therefore following error is returned
Solution —> download `libssl1.1` package from http://security.ubuntu.com/ubuntu/pool/main/o/openssl/libssl1.1_1.1.1f-1ubuntu2.12_amd64.deb and install it manually by running 

`sudo dpkg -i libssl1.1_1.1.1f-1ubuntu2.12_amd64.deb`

3. Install additional required modules:

```
sudo apt install php7.4-mysql, php7.4-mbstring, php7.4-redis, php7.4-imagick, php7.4-igbinary, php7.4-gmp, php7.4-bcmath, php7.4-curl, php7.4-gd, php7.4-zip, php7.4-imap, php7.4-ldap, php7.4-bz2, php7.4-ssh2, php-phpseclib, php7.4-common, php7.4-json, php7.4-xml, php7.4-dev, libsmbclient-dev, php-pear
```

4. Running below command to install the `php7.4-intl` package:

`apt install php7.4-intl`

returns an error:

```
The following packages have unmet dependencies:
 php7.4-intl : Depends: libicu66 (>= 66.1-1~) but it is not installable
```

Problem —> In Ubuntu 22.4 `libicu70` package is available
Solution —> download `libicu66` package from http://security.ubuntu.com/ubuntu/pool/main/i/icu/libicu66_66.1-2ubuntu2.1_amd64.deb and install it manually by running

`sudo dpkg -i libicu66_66.1-2ubuntu2.1_amd64.deb`

5. Use the following commands to install some common prerequisites:

```
sudo apt install smbclient
sudo apt install redis-server
sudo apt install unzip
sudo apt install openssl
sudo apt install rsync
sudo apt install imagemagick
```

6. Prevent the two manually installed packages from steps above to be upgraded when running the `apt upgrade` command by setting them on hold:

`sudo apt-mark hold libssl1.1 libicu66`

## WHERE Does This Need To Be Documented (Link)?

https://doc.owncloud.com/server/10.9/admin_manual/installation/manual_installation/server_prep_ubuntu_20.04.html#clean-ubuntu-20-04-installation

## WHY Should This Change Be Made?

Enterprise distros are not ready to deliver PHP 8.x soon therefore this is an attempt to have oC running with PHP 7.4 for Ubuntu 22.04 users

## (Optional) What Type Of Content Change Is This? 
- [x] New Content Addition
- [ ] Old Content Deprecation
- [ ] Existing Content Simplification
- [ ] Bug Fix to Existing Content

## (Optional) Which Manual Does This Relate To?
- [x] Admin Manual
- [ ] Developer Manual
- [ ] User Manual
- [ ] Android
- [ ] iOS
- [ ] Branded Clients
- [ ] Desktop Client
- [ ] Other

**Comment by crrodriguez** [comment]
You are trying to install focal packages on jamal that is never going to work..I will try myself and se ehow can help you.

**Comment by crrodriguez** [comment]
no PHP 7.4 is provided in the PPA.. I do not know if this is intentional or due to the fact ubuntu 22.04 is not released until tomorrow.

**Comment by crrodriguez** [comment]
it is not excluded intentionally but because it fails to build on jammy

`/<<PKGBUILDDIR>>/ext/openssl/openssl.c: In function ‘zm_startup_openssl’:
/<<PKGBUILDDIR>>/ext/openssl/openssl.c:1520:58: error: ‘RSA_SSLV23_PADDING’ undeclared (first use in this function); did you mean ‘RSA_PKCS1_PADDING’?
 1520 |         REGISTER_LONG_CONSTANT("OPENSSL_SSLV23_PADDING", RSA_SSLV23_PADDING, CONST_CS|CONST_PERSISTENT);`

And this failure happends because RSA_SSLV23_PADDING is no longer defined in openssl 3.x

**Comment by crrodriguez** [comment]
https://github.com/oerdnj/deb.sury.org/issues/1761 

this issue tracks progress.. stalled until that is done.

**Comment by mmattel** [comment]
Linking this closed PR from Nov. 2020 https://github.com/owncloud/core/pull/38094 ([WIP] php8.0)

**Comment by arcanisgk** [comment]
@pako81 add problem related to libapache2-mod-php7.4 in: https://github.com/oerdnj/deb.sury.org/issues/1767

**Comment by pako81** [comment]
Installation of PHP 7.4 on Ubuntu 22.04 LTS should be possible now, as https://github.com/oerdnj/deb.sury.org/issues/1761#issuecomment-1112209708 got fixed.

The following commands will install PHP 7.4 and all necessary extensions in order to have a basic oC installation running on Ubuntu 22.04 LTS + PHP 7.4 from ppa:ondrej repo.

```
sudo add-apt-repository ppa:ondrej/php -y

sudo apt update && sudo apt upgrade

sudo apt install php7.4 php7.4-intl php7.4-mysql php7.4-mbstring php7.4-redis php7.4-imagick php7.4-igbinary php7.4-gmp php7.4-bcmath php7.4-curl php7.4-gd php7.4-zip php7.4-imap php7.4-ldap php7.4-bz2 php7.4-ssh2 php-phpseclib php7.4-common php7.4-json php7.4-xml php7.4-dev libsmbclient-dev php-pear php7.4-apcu
```

**Comment by arcanisgk** [comment]
@pako81 Can you test if you have multiple versions of PHP to switch between it with:

a2dismod php7.2
a2enmod php7.4

**Comment by jnweiger** [comment]
I have tested 10.9.1 on Ubuntu 21.10 and Ubuntu 22.04 with ondrej's ppa -- works fine.
I'll use 22.04 as a normal test platform in future, and see what comes up.

My php7.4 install is based on the documented manual procedure for Ubuntu 18.04 and the recommendations given by Ondrej.  It currently looks like this: https://github.com/owncloud/QA/blob/master/tools/hetzner-deploy/deploy_oc10_apps.sh#L230

Ondrej recommends to also `add-apt-repository ppa:ondrej/apache2` -- I've tried with and without. Both works for me.
